sklearn/preprocessing/_data.py

```diff
@@ -123,7 +123,7 @@ def _handle_zeros_in_scale(scale, copy=True, constant_mask=None):
 @validate_params(
     {
         "X": ["array-like", "sparse matrix"],
-        "axis": [Options(Integral, {0, 1})],
+        "axis": [Interval(Integral, 0, 1, closed="both")],
```

Comment:
I'm not a maintainer but I remembered it was me that modified `sklearn.preprocessing.scale`. I believe that since `axis` can only be 0 or 1, directly using `Options` is more staightforward and comprehensible. If a user gives an invalid `axis` argument, I think he would like to see something like "you should use 0 or 1" instead of "you should use an integer between 0 and 1".

(Above is just my personal opinion.)

Reply:
Sorry about that was an unintentional edit. Your explanation makes sense. I will look further into it.